src/Symfony/Component/Console/Helper/ProgressBar.php

```diff
@@ -287,10 +288,21 @@ public function setProgress(int $step)
             $step = 0;
         }
 
-        $prevPeriod = (int) ($this->step / $this->redrawFreq);
-        $currPeriod = (int) ($step / $this->redrawFreq);
+        $prevStep = $this->step;
         $this->step = $step;
         $this->percent = $this->max ? (float) $this->step / $this->max : 0;
+
+        if (null === $this->redrawFreq) {
+            if (microtime(true) - $this->lastWriteTime >= .05 || $this->max === $step) {
+                $this->display();
+            }
+
+            return;
+        }
+
+        $prevPeriod = (int) ($prevStep / $this->redrawFreq);
```

Comment:
We could change this to: `intdiv($prevStep, $this->redrawFreq)`

Reply:
Sure, even though I tried to preserve current way of doing things

Reply:
On second thought, let's avoid doing unnecessary extra function calls